spock-core/src/main/java/org/spockframework/runtime/extension/builtin/StepwiseExtension.java

```diff
@@ -18,21 +18,45 @@
 import org.spockframework.runtime.extension.IAnnotationDrivenExtension;
 import org.spockframework.runtime.model.*;
 import org.spockframework.runtime.model.parallel.ExecutionMode;
+import spock.lang.Stepwise;
 
-import java.lang.annotation.Annotation;
 import java.util.List;
 
-public class StepwiseExtension implements IAnnotationDrivenExtension {
+public class StepwiseExtension implements IAnnotationDrivenExtension<Stepwise> {
   @Override
-  public void visitSpecAnnotation(Annotation annotation, final SpecInfo spec) {
+  public void visitSpecAnnotation(Stepwise annotation, final SpecInfo spec) {
     sortFeaturesInDeclarationOrder(spec);
     includeFeaturesBeforeLastIncludedFeature(spec);
     skipFeaturesAfterFirstFailingFeature(spec);
 
-    // Disable parallel child execution for @Stepwise tests
+    // Disable parallel child execution for @Stepwise specs
     spec.setChildExecutionMode(ExecutionMode.SAME_THREAD);
   }
 
+  @Override
+  public void visitFeatureAnnotation(Stepwise annotation, FeatureInfo feature) {
+    // Disable parallel iteration execution for @Stepwise features
+    feature.setExecutionMode(ExecutionMode.SAME_THREAD);
+
+    // If an error occurs in this feature, skip remaining iterations
```

Comment:
I would validate that this is indeed a data-driven spec and throw an exception otherwise.

Reply:
Why would you do that? The annotation does not make much sense on a non-data-driven feature, but it does not do any harm either. Arguably, there could be a compile error during AST analysis, if you really want to be strict with users, but a runtime error? I would be more lenient there, shall we? But if you insist, saying that Spock should enforce more structure or rules on users, I can do that. I am waiting for your feedback.

Reply:
BTW, `@Stepwise` on a spec with only a single feature method also does not yield any errors. I think that makes sense, because maybe during TDD the user wants to add more features later, has uncommented some temporarily or whatever. Similarly, we also should not enforce iterations on feature level for `@Stepwise` but simply handle the situation gracefully, which we already do: `skip` is set on the feature, but without any consequences, because there will be no next iteration if we are in the last one already. Having only 1 iteration is but a special case.

Reply:
The same problem s mentioned in https://github.com/spockframework/spock/pull/1437#issuecomment-1066028196 applies here, I tried. When throwing an exception here, it is **invisible in Maven**, even in debug mode. The test will be silently ignored without even reporting it as ignored. It is as if the test would not exist at all. So probably it is being sorted out by the JUnit platform while aggregating the test plan (or whatever the name for the group of tests to be run is).

The current behaviour is much better IMO. Exceptions thrown by extensions, at least annotation-driven ones (I did not test any global ones in this context yet) are impossible to debug for users like this. Looking at their build logs, they have no way of finding out what is happening. Probably they do not even notice, unless they explicitly look for the spec name in the log or they know the exact number of tests that should be run.

Reply:
Done in 47ceacc, but Maven problem is of course still unsolved, i.e. for now the situation is worse than before where we simply accepted annotations on non-parametrised features, because they had no effect anyway. I am actually feeling bad about this commit and would rather revert it, if you agree. Therefore, not marking this as resolved yet.

Reply:
After #1447 has now merged into the main branch, I think I am feeling OK about this change here, so I am resolving the review issue.